plutus-core/plutus-core/src/PlutusCore/DeBruijn/Internal.hs

```diff
@@ -183,8 +178,17 @@ getIndex :: (MonadReader Levels m, AsFreeVariableError e, MonadError e m) => Uni
 getIndex u = do
     Levels current ls <- ask
     case BM.lookup u ls of
-        Just ix -> pure $ levelToIndex current ix
+        Just ix -> pure $ levelToIx current ix
         Nothing -> throwing _FreeVariableError $ FreeUnique u
+  where
+    -- Compute the relative 'Index' of a absolute 'Level' relative to the current 'Level'.
+    -- `current` and `l` are influenced only by the spine of the AST,
```

Comment:
I think you could be a bit more specific than that: provided that the reader context is modified only using `declareBinder` and `withScope`, the current level will always be greater or equal to all the levels in the level mapping.

Reply:
Or even better: list it as an invariant of the `Levels` structure that the current level is greater than all levels in the mapping, and then point out in `declareBinder` and `withScope` that they maintain the invariant, and then you refer to it here.